modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -33,7 +33,8 @@ class {{classname}} {
         final String _path = '{{{path}}}'{{#pathParams}}.replaceAll('{' r'{{baseName}}' '}', {{{paramName}}}.toString()){{/pathParams}};
 
         final queryParams = <String, dynamic>{};
-        final headerParams = <String, dynamic>{
+        final headerParams = <String, dynamic>{ {{#httpUserAgent}}
```

Comment:
remove this space in the middle to avoid having trailing space in generated code

Reply:
I can't, mustache engine crashes without it...

Reply:
```{
{{#httpUserAgent}}'User-Agent': '{{{.}}}',{{/httpUserAgent}}
if (headers != null) ...headers,
```

Reply:
I'll generate an empty line in the final client then if `httpUserAgent` is not defined

Reply:
Yea had the same problem in some places and also settled with this appraoch.

Reply:
@agilob We could try something like this separately in another PR: https://stackoverflow.com/a/48133965/2553215